aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -862,7 +863,26 @@ internal class RealAWSCognitoAuthPlugin(
         onSuccess: Action,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        try {
+            val cognitoIdentityProviderClient = requireNotNull(
+                authEnvironment.cognitoAuthService.cognitoIdentityProviderClient
+            )
+
+            val appClient = requireNotNull(configuration.userPool?.appClient)
+
+            GlobalScope.launch {
+                ConfirmResetPasswordUseCase(cognitoIdentityProviderClient, appClient).execute(
+                    username,
+                    newPassword,
+                    confirmationCode,
+                    options,
+                    onSuccess,
+                    onError
+                )
+            }
+        } catch (ex: Exception) {
+            onError.accept(AuthException.InvalidUserPoolConfigurationException(ex))
```

Comment:
There are possibilities of multiple exceptions being thrown here, I'd expect the onError to propagate the appropriate error to the caller.

Reply:
Ideally, we would reach this catch block only if `requireNotNull` is not satisfied for `appClientId` or `cognitoIdentityProviderClient`. Which should not occur if all the configuration is correct.

All other exceptions (like ones returned by identity provider and remote API calls) are caught and boxed in onError passed to the use case and returned there.